pkg/frontend/mysql_cmd_executor.go

```diff
@@ -68,6 +68,10 @@ func parameterModificationInTxnErrorInfo() string {
 	return "Uncommitted transaction exists. Please commit or rollback first."
 }
 
+func abortTransactionErrorInfo() string {
+	return "Previous DML conflicts with existing constraints or data format. This transaction has to be aborted"
+}
+
 var (
 	errorOnlyCreateStatement        = moerr.NewInternalError(onlyCreateStatementErrorInfo())
```

Comment:
We do not use moerr.NewXXX as var.    We call moerr.NewXXX to create a new error when error happens, so that we will print a stack in log.

Reply:
good idea. i will replace it .